qa/vagrant/src/main/java/org/elasticsearch/packaging/test/PackagingTestCase.java

```diff
@@ -21,31 +21,54 @@
 
 import com.carrotsearch.randomizedtesting.JUnit3MethodProvider;
 import com.carrotsearch.randomizedtesting.RandomizedRunner;
+import com.carrotsearch.randomizedtesting.annotations.TestCaseOrdering;
 import com.carrotsearch.randomizedtesting.annotations.TestMethodProviders;
 import org.apache.commons.logging.Log;
 import org.apache.commons.logging.LogFactory;
+import org.elasticsearch.packaging.util.Distribution;
+import org.elasticsearch.packaging.util.Installation;
+import org.junit.Assert;
 import org.junit.Before;
+import org.junit.BeforeClass;
 import org.junit.Rule;
 import org.junit.rules.TestName;
 import org.junit.runner.RunWith;
 
+import static org.elasticsearch.packaging.util.Cleanup.cleanEverything;
+import static org.junit.Assume.assumeTrue;
+
 @RunWith(RandomizedRunner.class)
 @TestMethodProviders({
     JUnit3MethodProvider.class
 })
+@TestCaseOrdering(TestCaseOrdering.AlphabeticOrder.class)
 /**
  * Class that all packaging test cases should inherit from. This makes working with the packaging tests more similar to what we're
  * familiar with from {@link org.elasticsearch.test.ESTestCase} without having to apply its behavior that's not relevant here
  */
-public abstract class PackagingTestCase {
+public abstract class PackagingTestCase extends Assert {
 
     protected final Log logger = LogFactory.getLog(getClass());
 
     @Rule
     public final TestName testNameRule = new TestName();
 
     @Before
-    public void logTestNameBefore() {
+    public void setup() {
+        assumeTrue("only compatible distributions", distribution().packaging.compatible);
         logger.info("[" + testNameRule.getMethodName() + "]: before test");
     }
+
+    protected static Installation installation;
+
+    @BeforeClass
+    public static void cleanup() {
+        installation = null;
+        cleanEverything();
+    }
+
+    /** The {@link Distribution} that should be tested in this case */
+    protected abstract Distribution distribution();
```

Comment:
If we're adding this here, can we remove the abstract method declarations from `ArchiveTestCase` and `PackageTestCase`

Reply:
I don't see it in ArchiveTestCase anymore, but I've removed it now from PackageTestCase.